### DISCUSSION TITLE
Auto generate docs from typescript

### DISCUSSION THREAD

**Discussion by pustovalov** [storybookjs/storybook#15348] [Help]
Hey,

At the moment I'm using `react-docgen-typescript-loader` to automatically generate docs
Downloads last 30 days: [1.2m](https://yarnpkg.com/package/react-docgen-typescript-loader)

But this plugin is no longer supported and archived: https://github.com/strothj/react-docgen-typescript-loader, also doesn't work with `typescript ^4.3` (https://github.com/styleguidist/react-docgen-typescript/issues/356)

Is there any other way to automatically generate docs from TS?

**Comment by pustovalov** [comment]
Actually, I didn't notice any difference after deleting `react-docgen-typescript-loader` and using `react-docgen-typescript`

https://storybook.js.org/docs/ember/configure/typescript#mainjs-configuration


```
// .storybook/main.js

module.exports = {
  typescript: {
    check: false,
    checkOptions: {},
    reactDocgen: 'react-docgen-typescript',
    reactDocgenTypescriptOptions: {
      shouldExtractLiteralValuesFromEnum: true,
      propFilter: (prop) => (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true),
    },
  },
};
```

**Reply by shilman** [reply]
This is what we use under the hood https://www.npmjs.com/package/react-docgen-typescript-plugin
